### PR TITLE
refactor: remove parent task key

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Companion PR

- https://github.com/Substra/substra-backend/pull/554

## Description

- Remove `compute_task_parents`:
 - in `ComputeTask`
 - in gRPC communication
 - remove `expanded_compute_tasks`

## How has this been tested?

CI

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by ThibaultFy** [comment]
/e2e --refs substra-backend=feat/remove-parent_task_key --tests sdk,frontend --mode standalone

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]()
+
+## Removed
+
+- field `parent_task_keys` in `ComputeTask`
```
Comment (ThibaultFy) [inline CHANGELOG.md:11]: Put the PR number

File: lib/service/computetaskstate.go [hunk 2]
```diff
@@ -184,7 +184,7 @@ func (s *ComputeTaskService) propagateDone(triggeringParent, child *asset.Comput
 	}
 
 	// loop over parent, only change status if all parents are DONE
-	for _, parentKey := range child.ParentTaskKeys {
+	for _, parentKey := range GetParentTaskKeys(child.Inputs) {
```
Comment (ThibaultFy) [inline lib/service/computetaskstate.go:187]: Might worth testing if it take too long for big inputs (with 1000 datasamples for instance)

**Review by ThibaultFy — APPROVED** [review]
Thanks !!

**Comment by ThibaultFy** [comment]
/e2e --refs substra-backend=feat/remove-parent_task_key --tests substrafl --mode standalone

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by ThibaultFy** [comment]
/e2e --refs substra-backend=feat/remove-parent_task_key --tests NONE --mode standalone --benchmark mnist,camelyon

**Comment by Owlfred** [comment]
Something happened while processing your message: error: unknown option '--benchmark'
(Did you mean --benchmarks?)

**Comment by ThibaultFy** [comment]
/e2e --refs substra-backend=feat/remove-parent_task_key --tests NONE --mode standalone --benchmarks mnist,camelyon

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: lib/service/computetaskstate.go (hunk 2 above)
Comment (ThibaultFy) [reply lib/service/computetaskstate.go:187]: The e2e failes on SubstraFL and Benchmarks, seems to be because of a timeout error....

**Review by ThibaultFy — CHANGES_REQUESTED** [review]
E2E tests fail on SubstraFL and Benchmarks 😞

**Comment by ThibaultFy** [comment]
> End to end tests: ❌ [**FAILURE**](https://github.com/owkin/substra-ci/actions/runs/3823984016)
> 
> Jobs status:
> 
> * Tests Benchmark: [**❌**](https://github.com/owkin/substra-ci/actions/runs/3823984016/jobs/6505714234)
> * Tests Distributed: [**⏭️**](https://github.com/owkin/substra-ci/actions/runs/3823984016/jobs/6505714432)
> * Tests Standalone: [**✔️**](https://github.com/owkin/substra-ci/actions/runs/3823984016/jobs/6505714097)
> 
> Epic fail.

For the Mnist benchmark, the performances are impacted:
Expected: [0.8118, 0.7792, 0.9042, 0.8814, 0.9176, 0.9348, 0.9434, 0.9456, 0.9564, 0.9556]
Actual perf: [0.8118, 0.7792, 0.8388, 0.788, 0.869, 0.797, 0.8556, 0.8084, 0.866, 0.819]

**Comment by guilhem-barthes** [comment]
/e2e --refs substra-backend=feat/remove-parent_task_key --tests sdk,substrafl,frontend --benchmarks mnist,camelyon

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]()
+
+## Removed
+
+- field `parent_task_keys` in `ComputeTask` ([#112](https://github.com/Substra/orchestrator/pull/112))
+- view `expanded_compute_tasks` ([#112](https://github.com/Substra/orchestrator/pull/112))
```
Comment (ThibaultFy) [inline CHANGELOG.md:12]: You can just put #112, github does the link to the PR automatically :)

File: lib/service/computetaskstate.go [hunk 4]
```diff
@@ -204,7 +199,7 @@ func (s *ComputeTaskService) propagateDone(triggeringParent, child *asset.Comput
 			return nil
 		}
 	}
-	err := s.applyTaskAction(child, transitionTodo, fmt.Sprintf("Last parent task %s done", triggeringParent.Key))
+	err = s.applyTaskAction(child, transitionTodo, fmt.Sprintf("Last parent task %s done", triggeringParent.Key))
```
Comment (ThibaultFy) [inline lib/service/computetaskstate.go:202]: For my info: why remove the ":" here ?

**Review by ThibaultFy — APPROVED** [review]
Nice fix, can't wait to see it in action ;)

**Pull request merged**